### PR TITLE
rgw: support FQDN for RGW endpoint

### PR DESCRIPTION
**Pull request by parth-gr**
Now rgw can be pass by FQDN too,
to avoid problems where it is required
The format should be similar to
Ip(<IP>:<PORT> format), <FQDN>:<PORT> format

Signed-off-by: parth-gr <paarora@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/create-external-cluster-resources.py [hunk 1]
```diff
@@ -795,12 +795,25 @@ def create_rgw_admin_ops_user(self):
         jsonoutput = json.loads(output)
         return jsonoutput["keys"][0]['access_key'], jsonoutput["keys"][0]['secret_key']
 
+    def convert_fqdn_rgw_endpoint_to_ip(self,fqdn_rgw_endpoint):
+        try:
+            ipv4, port = fqdn_rgw_endpoint.split(':')
+        except ValueError:
+            raise ExecutionFailureException(
+            "Not a proper endpoint: {}, <IPv4>:<PORT>, format is expected".format(fqdn_rgw_endpoint))
```
[the remark this answers is not in the document]
Comment (parth-gr) [reply deploy/examples/create-external-cluster-resources.py]: Done

**Review by leseb — COMMENTED** [review]
How did you validate this?

**Comment by parth-gr** [comment]
> How did you validate this?

It's already been used for mon-endpoints, so the same functions will be applicable for RGW-endpoint for converting the fqdn to rgw-endpoint, right?

I can't have CI testing for it, because I need to map it to DNS, maybe it is possible?

**Comment by leseb** [comment]
> > How did you validate this?
> 
> It's already been used for mon-endpoints, so the same functions will be applicable for RGW-endpoint for converting the fqdn to rgw-endpoint, right?
> 
> I can't have CI testing for it, because I need to map it to DNS, maybe it is possible?
> 
> PS: Even I have read about sockets it does the work of listing the Ip through hostname

If we cannot have integration tests, can we have unit tests?

**Comment by parth-gr** [comment]
> > > How did you validate this?
> > 
> > 
> > It's already been used for mon-endpoints, so the same functions will be applicable for RGW-endpoint for converting the fqdn to rgw-endpoint, right?
> > I can't have CI testing for it, because I need to map it to DNS, maybe it is possible?
> > PS: Even I have read about sockets it does the work of listing the Ip through hostname
> 
> If we cannot have integration tests, can we have unit tests?

Added

**Review by leseb — APPROVED** [review]

**Comment by BlaineEXE** [comment]
I notice this commit message includes information about RHCS and OCS which are both downstream products. How does this affect upstream users? Let's keep commit messages and descriptions here to only upstream sentiments unless absolutely necessary.

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
Requesting changes to remove downstream-specific details from the commit message. I think it is better to keep Rook commits all in terms of upstream.

File: deploy/examples/create-external-cluster-resources.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def gen_arg_parser(cls, args_to_parse=None):
         output_group.add_argument("--rbd-data-pool-name", default="", required=False,
                                   help="Provides the name of the RBD datapool")
         output_group.add_argument("--rgw-endpoint", default="", required=False,
-                                  help="Rados GateWay endpoint (in <IP>:<PORT> format)")
+                                  help="Rados GateWay endpoint (in <IP>:<PORT> format). Note: FQDN is also supported(in <FQDN>:<PORT> format)")
```
Comment (BlaineEXE) [inline deploy/examples/create-external-cluster-resources.py:198]:
```suggestion
                                  help="RADOS Gateway endpoint (in <IP>:<PORT> format). Note: FQDN is also supported(in <FQDN>:<PORT> format)")
```

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**